packages/clay-form/CHECKBOX.mdx

```diff
@@ -32,22 +41,36 @@ Use [`checked`](#api-checked) property to check or not the checkbox.
 
 <Checkbox />
 
+## Advanced Usages
+
+Here you can see some options or ideas on how to extend our Checkbox component to fit your use.
+
+### Indeterminate Checkbox
+
 Use [`indeterminate`](#api-indeterminate) property for making the checkbox indeterminate.
 
 <CheckboxIndeterminate />
 
+### Checkbox with a callback
```

Comment:
This should likely be our first example since we expect users to always use these components in a controlled way